ASProgressPopUpView/ASProgressPopUpView.m

```diff
@@ -99,12 +100,24 @@ - (void)setPopUpViewColor:(UIColor *)color
     _popUpViewColor = color;
     _popUpViewAnimatedColors = nil; // animated colors should be discarded
     [self.popUpView setColor:color];
-    [self setGradientColors:@[color, color] withPositions:nil];
+    [self setGradientColors:@[color, color] withPositions:@[@(0.0), @(1.0)]];
 }
 
 - (void)setPopUpViewAnimatedColors:(NSArray *)colors
 {
-    [self setPopUpViewAnimatedColors:colors withPositions:nil];
+    if ([colors count]==0) {
+        return;
+    } else if ([colors count]==1) {
+        [self setPopUpViewAnimatedColors:@[colors[0], colors[0]] withPositions:@[@(0.0), @(1.0)]];
+    } else {
+        CGFloat spacer = 1.0 / ([colors count]-1);
+        NSMutableArray *positions = [[NSMutableArray alloc] initWithCapacity:[colors count]];
+        for (NSInteger i = 0; i < [colors count]; i++) {
+            [positions addObject:@(i * spacer)];
+        }
+        [positions addObject:@(1.0)];
```

Comment:
I've not had time to check, but I don't think it's necessary to manually calculate the gradient positions. If `nil` is passed as the arg to `positions` in the gradient layer, the color positions along the gradient are automatically calculated. (unless things have changed recently?)